api/src/middlewares/mongodb-sanitizer.ts

```diff
@@ -0,0 +1,37 @@
+import { NextFunction, Request, Response } from 'express';
+import * as _ from 'lodash';
+import { StatusCodes } from 'http-status-codes';
+
+const BAD_CHAR = '$';
+
+export const mongodbSanitizer = (req: Request, res: Response, next: NextFunction) => {
+	const body = req.body;
+
+	const val = hasBadCharacter(body, BAD_CHAR);
+
+	if (val) {
+		return res.status(StatusCodes.BAD_REQUEST).send({ error: `${BAD_CHAR} is not allowed as key prefix.` });
+	}
+
+	next();
+};
+
+const hasBadCharacter = (data: any, character: string): boolean => {
```

Comment:
Can you add also a null check (when data is null)

Reply:
It will return false since the two IF conditions are false so it is not necessary